homeassistant/components/darksky/weather.py

```diff
@@ -70,16 +73,22 @@
     }
 )
 
-MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=3)
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=DEFAULT_SCAN_INTERVAL)
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Dark Sky weather."""
+    global MIN_TIME_BETWEEN_UPDATES
+
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
     longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
     name = config.get(CONF_NAME)
     mode = config.get(CONF_MODE)
 
+    MIN_TIME_BETWEEN_UPDATES = config.get(CONF_SCAN_INTERVAL) or timedelta(
```

Comment:
We don't want the user to be able to override the throttle interval. That interval should be set by the developer to match the service level available from the api.

Reply:
That is the purpose of this change.  The Dark Sky API limits the number of free calls to 1000 per day.  The Dark Sky weather forecast is limited to either daily or hourly data even though both are returned in the same API call.  This is the issue raised in #31961. From architecture tickets related to displaying addition information returned from the Dark Sky API, the consensus seems to be keep the data consistent with other weather APIs and simple as so the data is always displayed the same and the [Weather Entity](https://developers.home-assistant.io/docs/entity_weather) does not need language translations.  Instead of creating a custom component, allowing the user to set the interval if they want to display both the hourly and daily forecast as well as tracking outside temperature/humidity using the Dark Sky sensor.

Reply:
The explanation @frenck provided in [the documentation PR](https://github.com/home-assistant/home-assistant.io/pull/12311#discussion_r389699852), this change may be counterproductive to the long term goals HA is trying to move towards.  Should I change this PR to simply change the Dark Sky API throttle to be 20 minutes?

Reply:
I don't think that will really solve the problem since the throttle is applied per platform and there's nothing stopping the user from creating multiple platforms with the same api key.

The best solution is the one I outlined above, to rewrite this integration in a major way. That is not a quick task though.

Reply:
Thanks and you are right, it wouldn't solve the problem.  I wanted to avoid rewriting the component as being new to the HA ecosystem I would likely make changes that would conflict with the Weather Entity and/or other weather components or conflict with other clean up work going on here or elsewhere in the project.  Although this seems like the proper course to take.  I will do more read through the link above and, time permitting, do a rewrite that would correct the underlying problem

Reply:
Sounds good.

Yes, a rewrite will likely mean a breaking change for this integration. But nothing outside the integration should be affected.